base/task.jl

```diff
@@ -197,6 +215,7 @@ function produce(v)
     t.state == :runnable || throw(AssertionError("producer.consumer.state == :runnable"))
     if empty
         schedule_and_wait(t, v)
+        ct = current_task() # When a task is copied, ct should be updated to new task ID.
```

Comment:
The reason I use `running` is that it is still running from the pov of the task itself. Without multithreading you won't be coping a actually running task anyway (unless you are copying yourself) and we don't currently support scheduling tasks onto threads.